<a href="https://colab.research.google.com/github/almsalahuddin/Data-Processing-and-Visualization/blob/main/data_viz_bike_share.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Quiz 4 # : CPSMA 4313 [Dr. Jacob's](https://github.com/nurfnick) Data processing visualization

## SQL Essentials

### A L M SALAHUDDIN

## Learning Some SQL with BigQuery

First I will try to authenticate in o0rder to use the dataset from goodle cloud platform.

In [2]:
from google.colab import auth
auth.authenticate_user()
print('Authenticated')

Authenticated


Above, it shows I authenticated. Now, I am able to use the dataset from BigQuery interface.

In [10]:
%%bigquery --project data-viz-bike-share
SELECT 
  COUNT(*) as total_trips
FROM `bigquery-public-data.austin_bikeshare.bikeshare_trips`

total_trips
0      1424786

The dataset I am using has total of 1424786 trips.

# Compute the average time for a trip based on starting points

In [11]:
%%bigquery --project data-viz-bike-share
SELECT 
  start_station_name,
  AVG(duration_minutes) as Average_time_for_trip
FROM `bigquery-public-data.austin_bikeshare.bikeshare_trips`
WHERE start_station_name is not null
GROUP BY start_station_name

start_station_name  Average_time_for_trip
0                         Zilker Park West              27.168259
1                  Toomey Rd @ South Lamar              28.115798
2          State Capitol @ 14th & Colorado              30.421842
3                         Waller & 6th St.              23.265843
4                               Pease Park              28.648889
..                                     ...                    ...
188  Republic Square @ Guadalupe & 4th St.              23.067313
189                             3rd & West              19.937466
190                               3rd/West              34.152547
191                          Nueces & 26th              16.805353
192                            26th/Nueces              24.521670

[193 rows x 2 columns]

 # Compute how many trips start at each starting point

In [21]:
%%bigquery --project data-viz-bike-share
SELECT 
  start_station_name,
  COUNT(*) as total_trips
FROM `bigquery-public-data.austin_bikeshare.bikeshare_trips`
GROUP BY start_station_name
ORDER BY total_trips DESC

start_station_name  total_trips
0        21st & Speedway @PCL        72799
1        Riverside @ S. Lamar        40635
2    City Hall / Lavaca & 2nd        36520
3              2nd & Congress        35307
4        Rainey St @ Cummings        34758
..                        ...          ...
188           Marketing Event            4
189             Eeyore's 2018            2
190                    Stolen            1
191             Eeyore's 2017            1
192     cesar Chavez/Congress            1

[193 rows x 2 columns]

In [19]:
%%bigquery --project data-viz-bike-share
SELECT 
  AVG(duration_minutes)
FROM `bigquery-public-data.austin_bikeshare.bikeshare_trips`

f0_
0  30.870428

Above, I calculated the average time for a trip.

In [20]:
%%bigquery --project data-viz-bike-share
SELECT duration_minutes, duration_minutes 
FROM `bigquery-public-data.austin_bikeshare.bikeshare_trips` 
LIMIT 1000

duration_minutes  duration_minutes_1
0                  39                  39
1                  31                  31
2                  31                  31
3                  30                  30
4                  19                  19
..                ...                 ...
995                33                  33
996                32                  32
997                 0                   0
998                 1                   1
999                80                  80

[1000 rows x 2 columns]